src/unittest/test.cpp

```diff
@@ -410,6 +410,52 @@ Markdown lol \\|\\|spoiler\\|\\| \\~\\~strikethrough\\~\\~ \\`small \\*code\\* b
 					}
 				});
 
+			set_test("USER_GET", false);
+			set_test("USER_GET_FLAGS", false);
+			if (!offline) {
+				bot.user_get(TEST_USER_ID, [](const dpp::confirmation_callback_t &event) {
+					if (!event.is_error()) {
+						auto u = std::get<dpp::user_identified>(event.value);
+						if (u.id == TEST_USER_ID) {
+							set_test("USER_GET", true);
+						} else {
+							set_test("USER_GET", false);
+						}
+						json j = json::parse(event.http_info.body);
+						uint64_t raw_flags = j["public_flags"];
+						if (j.contains("flags")) {
+							uint64_t flags = j["flags"];
+							raw_flags |= flags;
+						}
+						// testing all user flags from https://discord.com/developers/docs/resources/user#user-object-user-flags
+						if (
+								u.is_discord_employee() == 			((raw_flags & (1 << 0)) != 0) &&
+								u.is_partnered_owner() == 			((raw_flags & (1 << 1)) != 0) &&
+								u.has_hypesquad_events() == 		((raw_flags & (1 << 2)) != 0) &&
+								u.is_bughunter_1() == 				((raw_flags & (1 << 3)) != 0) &&
+								u.is_house_bravery() == 			((raw_flags & (1 << 6)) != 0) &&
+								u.is_house_brilliance() == 			((raw_flags & (1 << 7)) != 0) &&
+								u.is_house_balance() == 			((raw_flags & (1 << 8)) != 0) &&
+								u.is_early_supporter() == 			((raw_flags & (1 << 9)) != 0) &&
+								u.is_team_user() == 				((raw_flags & (1 << 10)) != 0) &&
+								u.is_bughunter_2() == 				((raw_flags & (1 << 14)) != 0) &&
+								u.is_verified_bot() == 				((raw_flags & (1 << 16)) != 0) &&
+								u.is_verified_bot_dev() == 			((raw_flags & (1 << 17)) != 0) &&
+								u.is_certified_moderator() == 		((raw_flags & (1 << 18)) != 0) &&
+								u.is_bot_http_interactions() == 	((raw_flags & (1 << 19)) != 0) &&
```

Comment:
i dont think we should hard code 1<< 18 etc into the code here. we should use the flags in `dpp/user.h`, no?
Our flags dont always match discord's flags and we can and do shuffle them about, which would break the unit tests.